frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/page_block/custom_page_block_component.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class CustomPageBlockComponentBuilder extends BlockComponentBuilder {
+  @override
+  BlockComponentWidget build(BlockComponentContext blockComponentContext) {
+    return CustomPageBlockComponent(
+      key: blockComponentContext.node.key,
+      node: blockComponentContext.node,
+      header: blockComponentContext.header,
+      footer: blockComponentContext.footer,
+    );
+  }
+}
+
+class CustomPageBlockComponent extends BlockComponentStatelessWidget {
+  const CustomPageBlockComponent({
+    super.key,
+    required super.node,
+    super.showActions,
+    super.actionBuilder,
+    super.configuration = const BlockComponentConfiguration(),
+    this.header,
+    this.footer,
+  });
+
+  final Widget? header;
+  final Widget? footer;
+
+  @override
+  Widget build(BuildContext context) {
+    final editorState = context.read<EditorState>();
+    final scrollController = context.read<EditorScrollController?>();
+    final items = node.children;
+
+    if (scrollController == null || scrollController.shrinkWrap) {
+      return SingleChildScrollView(
+        child: Builder(
+          builder: (context) {
+            final scroller = Scrollable.maybeOf(context);
+            if (scroller != null) {
+              editorState.updateAutoScroller(scroller);
+            }
+            return Column(
+              children: [
+                if (header != null) header!,
+                ...items.map(
+                  (e) => Container(
+                    constraints: BoxConstraints(
+                      maxWidth:
+                          editorState.editorStyle.maxWidth ?? double.infinity,
+                    ),
+                    padding: editorState.editorStyle.padding,
+                    child: editorState.renderer.build(context, e),
+                  ),
+                ),
+                if (footer != null) footer!,
+              ],
+            );
+          },
+        ),
+      );
+    } else {
+      int extentCount = 0;
+      if (header != null) extentCount++;
+      if (footer != null) extentCount++;
+
+      return ScrollablePositionedList.builder(
+        shrinkWrap: scrollController.shrinkWrap,
+        itemCount: items.length + extentCount,
+        itemBuilder: (context, index) {
+          editorState.updateAutoScroller(Scrollable.of(context));
+          if (header != null && index == 0) {
+            return IgnoreEditorSelectionGesture(
+              child: header!,
+            );
+          }
+
+          if (footer != null && index == (items.length - 1) + extentCount) {
+            return IgnoreEditorSelectionGesture(
+              child: footer!,
+            );
+          }
+
+          final childNode = items[index - (header != null ? 1 : 0)];
+          final isOverflowType = overflowTypes.contains(childNode.type);
```

Comment:
I think it might not be the proper way to handle the overflow logic here. 

It seems it can only handle the blocks at the first level. If we have databases or tables at a level **greater than 1**, then it can't handle overflow in the current implementation, right?

Reply:
Good point. I'll test the database at a high level and think about how to handle it within the overflow type

Reply:
I have a basic idea: removing the padding in the editor configuration and adding extra padding to each block. We can discuss it tomorrow.